sdk/nodejs/storage/zMixins.ts

```diff
@@ -401,27 +419,68 @@ export class QueueEventSubscription extends appservice.EventSubscription<QueueCo
     }
 }
 
+function merge(trigger: appservice.BindingDefinition, appSettings: pulumi.Input<{ [key: string]: string }>, otherBindings?: appservice.BindingSettings[]) {
+    if (otherBindings) {
+        const bindings = pulumi.all(otherBindings.map(b => b.binding)).apply(bs => [trigger, ...bs]);
+
+        const perFunctionSettings = otherBindings.map(b => b.settings);
+        const settings =  pulumi.all([appSettings, ...perFunctionSettings]).apply(items => items.reduce((a, b) => ({ ...a, ...b }), {}));
+        return { bindings, settings };
+    } else
+    {
+        return { bindings: [trigger], settings: appSettings };
+    }
+}
+
 /**
  * Azure Function triggered by a Storage Queue.
  */
-export class QueueFunction extends appservice.FunctionBase<QueueContext, Buffer, void> {
+export class QueueFunction extends appservice.FunctionBase<QueueContext, Buffer, void | Record<string, any>> {
     constructor(name: string, args: QueueFunctionArgs) {
         const bindingConnectionKey = pulumi.interpolate`${args.queue.storageAccountName}ConnectionStringKey`;
 
         const account = pulumi.all([args.queue.resourceGroupName, args.queue.storageAccountName])
             .apply(([resourceGroupName, storageAccountName]) =>
                 storage.getAccount({ resourceGroupName, name: storageAccountName }));
 
-        const appSettings = pulumi.all([account.primaryConnectionString, bindingConnectionKey]).apply(
+        const connectionStingsSetting = pulumi.all([account.primaryConnectionString, bindingConnectionKey]).apply(
             ([connectionString, key]) => ({ [key]: connectionString }));
 
-        super(name, [<QueueBindingDefinition>{
+        const trigger: QueueBindingDefinition = {

```

Comment:
The `QueueBindingDefinition` should now be called `QueueInputBindingDefinition` since there now also is an output binding

Reply:
Renamed to `QueueTriggerBindingDefinition` (and `QueueBindingDefinition` is the base interface now).

Reply:
You are right - input would also be wrong, its a trigger.